File: TripKitAndroid/src/main/java/com/skedgo/android/tripkit/CarPark.java
package com.skedgo.android.tripkit;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@TypeAdapters
@Immutable
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersCarPark.class)
public abstract class CarPark {
  public abstract String identifier();
  public abstract String name();
  public abstract int totalSpaces();
  public abstract int availableSpaces();
  public abstract long lastUpdate();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/CarPark.kt
package com.skedgo.tripkit

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@TypeAdapters
@Immutable
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersCarPark::class
)
abstract class CarPark {
    abstract fun identifier(): String

    abstract fun name(): String

    abstract fun totalSpaces(): Int

    abstract fun availableSpaces(): Int

    abstract fun lastUpdate(): Long
}